### ISSUE TITLE
Variables {{.AccessKey}} and {{.SecretKey}} are not being interpolated in bundle_upload_command

### ISSUE THREAD

**Issue by sobotklp** [mitchellh/packer#2496]
I noticed this issue in Packer 0.8.1 and 0.8.2

My packer build file includes this line:
```
"bundle_upload_command": "sudo EC2_AMITOOL_HOME={{user `ec2_amitool_home`}} -n {{user `ec2_amitool_home`}}/bin/ec2-upload-bundle -b {{.BucketName}} -m {{.ManifestPath}} -a {{.AccessKey}} -s {{.SecretKey}} -d {{.BundleDirectory}} --region {{user `region`}} --batch --retry",
```

When running packer with the -debug flag, I see this output:
```
1437685192,,ui,say,==> paravirtual.instance-store: Running: sudo EC2_AMITOOL_HOME=/usr/local/ec2/ec2-ami-tools -n /usr/local/ec2/ec2-ami-tools/bin/ec2-upload-bundle -b my-packer-images -m /dev-INSTANCE-release-1.8.3_1_1437684721.manifest.xml -a  --secret  -d /tmp --region us-west-2 --batch --retry
```

It appears that the {{.AccessKey}} and {{.SecretKey}} fields are no longer being written into the command. My AWS credentials are being read from environment variables, currently. I have tried to pass the values explicitly using the "access_key" and "secret_key" variables, but I get the same result.

Finally, _ec2-bundle-vol fails_ with this output:
```
1437685192,,ui,message,    paravirtual.instance-store: missing mandatory parameter: --secret-key
1437685193,,ui,message,    paravirtual.instance-store: Try 'ec2-upload-bundle --help'
1437685193,,ui,error,==> paravirtual.instance-store: Bundle upload failed. Please see the output above for more\n==> paravirtual.instance-store: details on what went wrong.
```

**Comment by sobotklp** [comment]
This command worked just fine with packer 0.7.5 and the same version of ec2_ami_tools

**Comment by cbednarski** [comment]
Thanks for the report! We reworked interpolation in 0.8 and some cases were overlooked.

**Comment by empath** [comment]
Is there a work around for this?

**Comment by cbednarski** [comment]
@empath I have not had time to look into this yet. If this field is not interpolated at all, no. However if `{{ user \`variables\` }}` work you can use those in place of `{{ .AccessKey }}`, for example.

**Comment by cbednarski** [comment]
I believe this is fixed in #2596. You can test out the patch by building packer from master. If you are still having issues please feel free to reopen or open another issue. Thanks!
